x/wasm-storage/keeper/abci.go

```diff
@@ -13,17 +13,30 @@ import (
 	"github.com/sedaprotocol/seda-wasm-vm/tallyvm"
 )
 
-func (k Keeper) EndBlock(ctx sdk.Context) error {
-	err := k.ProcessExpiredWasms(ctx)
+func (k Keeper) EndBlock(ctx sdk.Context) (err error) {
+	// Use defer to prevent returning an error, which would cause
+	// the chain to halt.
+	defer func() {
+		// Handle a panic.
+		if r := recover(); r != nil {
+			k.Logger(ctx).Error("recovered from panic in wasm-storage EndBlock", "err", r)
+		}
+		// Handle an error.
+		if err != nil {
+			k.Logger(ctx).Error("error in wasm-storage EndBlock", "err", err)
+		}
+		err = nil
```

Comment:
Why not propagate the error up to the chain? What happens if we don't set err to nil?

Reply:
The chain halts :)